### PR TITLE
Bug 1793267: Add -mod=vendor fallback for `go list` 

### PR DESCRIPTION
**Pull request by soltysh**
This mostly picks https://github.com/openshift/library-go/pull/697, but additionally I picked all of api & client-go to match current levels.

/assign @tnozicka

### CODE REVIEW

File: pkg/helpers/describe/describer.go [hunk 1]
```diff
@@ -1776,12 +1802,34 @@ func (d *HostSubnetDescriber) Describe(namespace, name string, settings describe
...
 		formatString(out, "Pod Subnet", hs.Subnet)
-		formatString(out, "Egress CIDRs", strings.Join(hs.EgressCIDRs, ", "))
-		formatString(out, "Egress IPs", strings.Join(hs.EgressIPs, ", "))
+		formatString(out, "Egress CIDRs", hostCIDRsJoin(hs.EgressCIDRs, ", "))
+		formatString(out, "Egress IPs", hostIPsJoin(hs.EgressIPs, ", "))
 		return nil
 	})
 }
 
+func hostCIDRsJoin(cidrs []networkv1.HostSubnetEgressCIDR, sep string) string {
```
Comment (tnozicka) [inline pkg/helpers/describe/describer.go]: pls add simple table driven unit test

File: hack/verify-generated-completions.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ trap "cleanup" EXIT
 
 COMPLETION_ROOT_REL="contrib/completions"
 COMPLETION_ROOT="${OS_ROOT}/${COMPLETION_ROOT_REL}"
-TMP_COMPLETION_ROOT_REL="_output/verify-generated-completions/"
+TMP_COMPLETION_ROOT_REL="_output/verify-generated-completions"
```
Comment (tnozicka) [inline hack/verify-generated-completions.sh]: I suppose I'd like a commit failing that verification so we are sure, but that can be a followup PR

File: go.mod [hunk 3]
```diff
@@ -45,17 +45,17 @@ require (
...
-	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
+	golang.org/x/crypto v0.0.0-20191011191535-87dc89f01550
+	golang.org/x/net v0.0.0-20191209160850-c0dbc17a3553
 	golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456
 	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
 	gopkg.in/ldap.v2 v2.5.1
-	k8s.io/api v0.17.0
-	k8s.io/apimachinery v0.17.0
-	k8s.io/apiserver v0.17.0
+	k8s.io/api v0.17.1
```
Comment (tnozicka) [inline go.mod]: replaces are negating this and other changes

File: hack/verify-generated-completions.sh (hunk 2 above)
Comment (soltysh) [reply hack/verify-generated-completions.sh]: Failure happened in this PR, that's why I'm fixing it.

File: go.mod (hunk 3 above)
Comment (soltysh) [reply go.mod]: Fixed the replace directives.

**Comment by soltysh** [comment]
Rebased to resolve problems. Tagging back.

**Pull request merged**